src/components/trade/tradeWidget/useBuySell.ts

```diff
@@ -112,33 +112,25 @@ export const useBuySell = ({
   const liquidityQuery = useGetTradeLiquidity(source.address, target.address);
 
   const checkLiquidity = () => {
-    const checkSource = () => {
-      if (sourceInput === '' || sourceInput === '...') {
-        return false;
-      }
+    const checkSource = () =>
+      new SafeDecimal(sourceInput).gt(maxSourceAmountQuery.data || 0);
 
-      return new BigNumber(sourceInput).gt(maxSourceAmountQuery.data || 0);
-    };
-
-    const checkTarget = () => {
-      if (targetInput === '' || targetInput === '...') {
-        return false;
-      }
-
-      return new BigNumber(targetInput).gt(liquidityQuery.data || 0);
-    };
+    const checkTarget = () =>
+      new SafeDecimal(targetInput).gt(liquidityQuery.data || 0);
 
     const set = () => setIsLiquidityError(true);
     setIsLiquidityError(false);
 
     if (isTradeBySource) {
       if (checkSource()) {
         setTargetInput('...');
+        setRate('');
```

Comment:
why are we mixing additional things in the PR?

Reply:
Removed